File: frontend/src/types/debt.ts
export interface Debt {
  id: number;
  name: string;
  type: 'advance' | 'personal' | 'credit_card' | 'loan' | 'bill_arrears' | 'medical' | 'other';
  creditor: string;
  original_amount: number;
  current_balance: number;
  minimum_payment: number;
  interest_rate: number;
  due_day: number | null;
  priority: number;
  is_auto_deduct: number;
  notes: string | null;
  status: 'active' | 'paid_off' | 'paused';
  total_paid: number;
  created_at: string;
}

export interface DebtPayment {
  id: number;
  debt_id: number;
  amount: number;
  date: string;
  note: string | null;
}

export interface DebtSummary {
  total_debts: number;
  total_owed: number;
  total_minimum_monthly: number;
  monthly_interest_cost: number;
  debts_by_priority: Debt[];
}

export interface PayoffPlan {
  strategy: string;
  debts: Array<{
    id: number;
    name: string;
    current_balance: number;
    interest_rate: number;
    estimated_payoff_date: string;
    months_to_payoff: number;
    total_interest: number;
    total_interest_paid: number;
  }>;
  total_months: number;
  total_interest: number;
  debt_free_date: string;
}

export interface PaycheckAllocation {
  paycheck_amount: number;
  allocations: Array<{
    name: string;
    amount: number;
    type: 'auto_deduct' | 'housing' | 'utility' | 'minimum_payment' | 'essentials' | 'extra_payment' | 'buffer';
    priority: number;
    running_remaining: number;
    debt_id?: number;
  }>;
  total_allocated: number;
  remaining: number;
  shortfall: number;
  essentials_covered: boolean;
  warnings: string[];
}

File: backend/app/models/debt.py
from pydantic import BaseModel, Field
from typing import Optional, Literal, List


class DebtCreate(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    type: Literal['advance', 'personal', 'credit_card', 'loan', 'bill_arrears', 'medical', 'other']
    creditor: str = Field(min_length=1, max_length=200)
    original_amount: int = Field(gt=0)
    current_balance: int = Field(ge=0)
    minimum_payment: int = Field(default=0, ge=0)
    interest_rate: float = Field(default=0, ge=0, le=100)
    due_day: Optional[int] = Field(None, ge=1, le=31)
    priority: int = Field(default=5, ge=1, le=10)
    is_auto_deduct: bool = False
    notes: Optional[str] = None


class DebtUpdate(BaseModel):
    name: Optional[str] = None
    creditor: Optional[str] = None
    current_balance: Optional[int] = Field(None, ge=0)
    minimum_payment: Optional[int] = Field(None, ge=0)
    interest_rate: Optional[float] = Field(None, ge=0, le=100)
    due_day: Optional[int] = Field(None, ge=1, le=31)
    priority: Optional[int] = Field(None, ge=1, le=10)
    is_auto_deduct: Optional[bool] = None
    notes: Optional[str] = None
    status: Optional[Literal['active', 'paid_off', 'paused']] = None


class DebtPaymentCreate(BaseModel):
    amount: int = Field(gt=0)
    date: str = Field(pattern=r'^\d{4}-\d{2}-\d{2}$')
    note: Optional[str] = None


class PaycheckAllocation(BaseModel):
    paycheck_amount: int = Field(gt=0)
    pay_date: str = Field(pattern=r'^\d{4}-\d{2}-\d{2}$')
